src/shared_types.rs

```diff
@@ -45,17 +45,8 @@ impl DefaultPyErr for IntegratedOperationException {
 }
 
 #[pyclass]
-#[derive(Clone, Copy)]
-pub struct ShallowSubscription(pub SubscriptionId);
-#[pyclass]
-#[derive(Clone, Copy)]
-pub struct DeepSubscription(pub SubscriptionId);
-
-#[derive(FromPyObject)]
-pub enum SubId {
```

Comment:
Unsure if folding unobserve/unobserve_deep calls was a workaround previously, but it seems unnecessary now.